### ISSUE TITLE
Possible bug due to Camouflager, Time Master, Medic Shield combination

### ISSUE THREAD

**Issue by NyphoX** [Eisbison/TheOtherRoles#83]
We've just encountered the following issue and from the role descriptions it seems like this isn't intended behaviour.

The Medic shields the Time Master. Later in the same round, the Time Master uses his shield. A second or so later the Camouflager uses his ability and kills the Time Master within the time frame of his shield during the Camouflage. The Medic is still alive at that point.

For the Camouflager the description says "The shield is not visible anymore.", but doesn't state that it gets removed (for Medic or Time Master both).

As I understood the Time Master, the whole role is based on outplaying the Impostor by timing the shield right. However, it seems that the Camouflage "removes" the Medic's shield and the Time Master's self-shield (or a combination of these three). I believe this is not the desired behaviour, as it basically makes the Time Master useless whenever there is a Camouflager in the game.

Concerning the medic: the shield is not visible (so that work's correct), but shouldn't it also stay active even during the Camouflage (assuming it is not a bug)?

**Comment by Eisbison** [comment]
Thank you for your bug report. What happened there is definitely not the intended behaviour. Camouflage is just about  visuals, the shields should still remain active. I'll look into things later today 👍

**Comment by Eisbison** [comment]
I tried every combination now and things seemed to work as intended. Should we close this for now and reopen it in case someone runs into this problem again?

**Comment by Eisbison** [comment]
Could it be that there was an Eraser/Shifter/Sidekick that changed the role of the Medic?

**Comment by NyphoX** [comment]
Hm, that is weird. There was certainly no Eraser, Shifter or Sidekick present. If I remember correctly, the entire situation was like this:

- 5 players
- 1 Impostor Role (Vampire or Camouflager both at 10%, Camouflager was chosen)
- 3 or 3-4 Crewmate/Neutral Roles (Medic, Time Master, Sheriff, Engineer, Lighter, Snitch)
- All roles at equal probability of 10%
- Medic shield granted maybe 30 sec (a long time) ahead of the situation
- Time Master shield activated 1-2 sec before the kill (3 sec shield duration)
- Camouflage activate ~1 sec before the kill
- All 3 roles involved were in a small area during the kill (Security-Reactor-crossing on Skeld)

Your call to close this, if it's not reproducible. I could help if you want to try to reproduce this once more.

**Comment by EoF-1141** [comment]
There was a similar report of child being able to die during camo
Probably related

**Comment by Eisbison** [comment]
@EoF-1141  Could be, if handleMurderAttempt is affected in some ways

**Comment by Lunaresk** [comment]
I also ran in a similar situation where I, as an Impostor, was able to kill someone with a medic shield while the medic was still alive. I can't remember everything, but we had a Medic, Sheriff and a Lovers couple in there (plus more I won't remember), played on Airship and when I'm right I had the vanilla Impostor role. I don't know which role my Impostorbuddy had, cause he died to the Sheriff on the first round.
In my situation I killed the shielded person **without camo being active**.
